CMakeLists.txt

```diff
@@ -91,6 +91,23 @@ if (ENABLE_HARDENING)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${HARDENING_LDFLAGS}")
 endif()
 
+option(DEV_BUILD "Enable additional build options to simplify debugging" OFF)
+
+# Assuming the C and CXX compiler IDs are equal.
+# For clang `MATCHES` is used to match both Clang and AppleClang.
+if (CMAKE_C_COMPILER_ID MATCHES "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "10" OR
+    CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "8")
+    add_compile_flags("C;CXX" "-fmacro-prefix-map=${CMAKE_SOURCE_DIR}=.")
+endif()
+
+# Assuming the C and CXX compiler IDs are equal.
+# For clang `MATCHES` is used to match both Clang and AppleClang.
+if (NOT DEV_BUILD AND
+    CMAKE_C_COMPILER_ID MATCHES "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "3.8" OR
+    CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "4.4.7")
+    add_compile_flags("C;CXX" "-fdebug-prefix-map=${CMAKE_SOURCE_DIR}=.")
```

Comment:
BTW these flags are not propagated to libunwind:
```
$ readelf --debug-dump=info ./build/src/tarantool | grep src
...
    <81dde0>   DW_AT_name        : (indirect string, offset: 0xbdcfc): /home/iverbin/git/tarantool/third_party/libunwind/src/mi/strerror.c
    <81dde4>   DW_AT_comp_dir    : (indirect string, offset: 0xbde00): /home/iverbin/git/tarantool/build/build/libunwind/work/src
    <81ed13>   DW_AT_name        : (indirect string, offset: 0xbdfdc): /home/iverbin/git/tarantool/third_party/libunwind/src/mi/Gget_accessors.c
...
```

Probably static build is also affected.

Reply:
Fixed: added these flags to new global variables used for passing compiler flags to dependencies. Also added these flags to static build.